tests/agent_tests/agent_tests.cc

```diff
@@ -78,3 +79,109 @@ TEST_P(AgentTests, SetThenGetEconParam) {
+
+  EXPECT_NEAR(A, agent_->PV(n,0,0,A), finance_eps); // discount rate = 0
+  EXPECT_NEAR(0, agent_->PV(0,i,0,A), finance_eps);  // discount time = 0
+  EXPECT_NEAR(A / (1+i), agent_->PV(1,i,0,A), finance_eps); // discount time = 1
+  EXPECT_NEAR(A * (1 - std::pow((1 + i), -n)) / i, agent_->PV(n,i,0,A), finance_eps); // general
+  EXPECT_NEAR(A * (1 - std::pow((1 + (i/12)), -n*12)) / (i/12), agent_->PV(n*12,i/12,0,A), finance_eps); // general
+  EXPECT_NEAR(A * (1 - std::pow((1 + i), n)) / i, agent_->PV(-n,i,0,A), finance_eps); // negative time
+
+  EXPECT_NEAR(F + A, agent_->PV(n,0,F,A), finance_eps); // discount rate = 0
+  EXPECT_NEAR(F, agent_->PV(0,i,F,A), finance_eps); // discount time = 0
+  EXPECT_NEAR((F + A) / (1+i), agent_->PV(1,i,F,A), finance_eps); // discount time = 1
+  EXPECT_NEAR(F / std::pow((1+i), n) + A * (1 - std::pow((1 + i), -n)) / i,
+              agent_->PV(n,i,F,A), finance_eps); // general
+
+}
+
+
+TEST_P(AgentTests, FV) {
+
+  int n = 5;
+  double i = 0.1;
+  double P = 1;
+  double A = 0.5;
+  double finance_eps = 1e-4;
+
+  // testing by "round trip" with already tested functions
+  EXPECT_NEAR(P, agent_->PV(n,0,agent_->FV(n,0,P,0),0), finance_eps);  // discount rate = 0
+  EXPECT_NEAR(P, agent_->PV(0,i,agent_->FV(0,i,P,0),0), finance_eps);  // discount time = 0
+  EXPECT_NEAR(P, agent_->PV(i,i,agent_->FV(1,i,P,0),0), finance_eps);  // discount time = 1
+  EXPECT_NEAR(P, agent_->PV(n,i,agent_->FV(n,i,P,0),0), finance_eps);  // general
+  EXPECT_NEAR(P, agent_->PV(n*12,i/12,agent_->FV(n*12,i/12,P,0),0), finance_eps);  // general
+  EXPECT_NEAR(P, agent_->PV(-n,i,agent_->FV(-n,i,P,0),0), finance_eps);  // general
+  
+  // manual testing
+  EXPECT_NEAR(A, agent_->FV(n,0,0,A), finance_eps); // discount rate = 0
+  EXPECT_NEAR(0, agent_->FV(0,i,0,A), finance_eps);  // discount time = 0
+  EXPECT_NEAR(A, agent_->FV(1,i,0,A), finance_eps); // discount time = 1
+  EXPECT_NEAR(A * (std::pow((1 + i), n) - 1) / i, agent_->PV(n,i,0,A), finance_eps); // general
+  EXPECT_NEAR(A * (std::pow((1 + i/12), n*12) - 1) / i/12, agent_->PV(n*12,i/12,0,A), finance_eps); // general
+  EXPECT_NEAR(A * (std::pow((1 + i), -n) - 1) / i, agent_->PV(-n,i,0,A), finance_eps); // general
+
+  EXPECT_NEAR(P + A, agent_->FV(n,0,P,A), finance_eps); // discount rate = 0
+  EXPECT_NEAR(P, agent_->FV(0,i,P,A), finance_eps); // discount time = 0
+  EXPECT_NEAR(P * (1+i) + A, agent_->FV(1,i,P,A), finance_eps); // discount time = 1
+  EXPECT_NEAR(P * std::pow((1+i), n) + A * (std::pow((1 + i), n) - 1),
+              agent_->FV(n,i,P,A), finance_eps); // general
+
+}
+
+TEST_P(AgentTests, PMT) {
+
+  int n = 5;
+  double i = 0.1;
+  double P = 1.5;
+  double F = 3.5;
+  double finance_eps = 1e-4;
+
+  // all testing by "round trip" with already tested functions
+
+  EXPECT_NEAR(P, agent_->PV(n,0,0,agent_->PMT(n,0,P,0)), finance_eps); // discount rate = 0
+  EXPECT_NEAR(P, agent_->PV(0,i,0,agent_->PMT(0,i,P,0)), finance_eps); // discount time = 0
```

Comment:
The left hand side evaluates to `nan` - this case may not be well defined and we should discuss the right solution.

Basically, there is no correct answer for how big the equivalent payment is when there are 0 payment.  So maybe this is a case where we want to make sure n >= 0 and otherwise throw an exception.